staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/dynamic_cafile_content.go

```diff
@@ -158,17 +159,45 @@ func (c *DynamicFileCAContent) Run(workers int, stopCh <-chan struct{}) {
 	// doesn't matter what workers say, only start one.
 	go wait.Until(c.runWorker, time.Second, stopCh)
 
-	// start timer that rechecks every minute, just in case.  this also serves to prime the controller quickly.
-	go wait.PollImmediateUntil(FileRefreshDuration, func() (bool, error) {
-		c.queue.Add(workItemKey)
-		return false, nil
-	}, stopCh)
-
-	// TODO this can be wired to an fsnotifier as well.
+	// start the loop that watches the CA file until stopCh is closed.
+	go wait.NonSlidingUntil(func() { c.watchCAFile(stopCh) }, time.Second, stopCh)
```

Comment:
Since `watchCAFile` only returns on error, let's make the retry every minute instead of every second to avoid warm loops.

Reply:
Why non-sliding?  I think we want the timer to interval to start when the function ends, not when it starts.

Reply:
Thanks for the review. `watchCAFile` also returns on file removal. The 1 second interval and non-sliding is for the mounted volume case: if a secret includes the CA bundle is mounted in a Pod, updating the secret leads to a file replacement, not content update. fsnotify watcher will receive a `fsnotify.Remove` event of the old file. If we use one minute as the interval and slidingUntil, it will restart the watch after one minute even the new file is ready immediately.
Maybe I could handle file removal in `watchCAFile` and start a new watch immediately without relying on the external loop. Then the external loop handles error case only and can wait longer interval.

Reply:
I have made the change, PTAL.